WcaOnRails/app/controllers/results_submission_controller.rb

```diff
@@ -8,20 +8,48 @@ class ResultsSubmissionController < ApplicationController
 
   def new
     @competition = competition_from_params
-    @results_submission = ResultsSubmission.new
+    @results_validator = CompetitionResultsValidator.new(@competition.id)
+  end
+
+  def upload_json
+    @competition = competition_from_params
+    if @competition.results_submitted?
+      return redirect_to competition_submit_results_edit_path
+    end
+    # Do json analysis + insert record in db, then redirect to check inbox
+    # (and delete existing record if any)
+    upload_json_params = params.require(:upload_json).permit(:results_file)
+    upload_json_params[:competition_id] = @competition.id
+    @upload_json = UploadJson.new(upload_json_params)
+
+    # This makes sure the json structure is valid!
+    if @upload_json.import_to_inbox
+      flash[:success] = "JSON File has been imported."
+      @competition.uploaded_jsons = [UploadedJson.create(json_str: @upload_json.results_json_str)]
```

Comment:
This is actually part of a long term plan of registering all submitted jsons as well as their errors/warnings.
Since everything to do this will not be part of this PR, I thought that making sure the table only contains the last submitted json would be better.

It should be fine though to append it though, as `UploadedJson.last` will always return the one we want.